department-of-veterans-affairs/abd-vro
# Basic Image Build and Container Fidelity Checks

## What was the problem?

* Some directories were structured in a way which did not align with others in the repo
* There was no action to run basic image building and validating container health checks as a sort of proto E2E testing across the whole platform. 

Associated tickets or Slack threads:

* https://github.com/department-of-veterans-affairs/abd-vro/issues/1638
* https://github.com/department-of-veterans-affairs/abd-vro/issues/1641

## How does this fix it?[^1]

* Adds a new GitHub workflow which runs on PRs which are marked ready for review on select branches that does the Docker steps mentioned above.

## How to test this PR
- Successful action which builds images, runs docker compose, and validates healthchecks 


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
